source/php-frameworks/symfony.txt

```diff
@@ -0,0 +1,397 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Run the following commands to enter your project directory and install
+MongoDB:
+
+.. code-block:: bash
+
+   cd restaurants
+   pecl install mongodb
+
+Install PHP Driver and Doctrine ODM
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Add the MongoDB PHP driver and the MongoDB Doctrine ODM bundle to your
+application. The bundle integrates the ODM library into Symfony so that
+you can read from and write objects to MongoDB. To learn more, see the
+:ref:`php-symfony-resources` section of this guide.
+
+Run the following commands to install the driver and the ODM:
+
+.. code-block:: bash
+
+   composer require mongodb/mongodb
+   composer require doctrine/mongodb-odm-bundle
+
+.. tip::
+
+   After running the preceding commands, you might encounter the
+   following prompt:
+
+   .. code-block:: bash
+      :copyable: false
+
+      Do you want to execute this recipe?
+
+   Select ``yes`` from the response options to add the ODM library to your
+   application.
+
+To ensure that the Doctrine ODM is enabled in your project, verify that your
+``config/bundles.php`` file contains the highlighted entry in the
+following code:
+
+.. code-block:: php
+   :caption: config/bundles.php
+   :emphasize-lines: 3
+   
+   return [
+       // ...
+       Doctrine\Bundle\MongoDBBundle\DoctrineMongoDBBundle::class => ['all' => true],
+   ];
+
+Configure the ODM
+`````````````````
+
+In the ``config/packages`` directory, replace the contents of your
+``doctrine_mongodb.yaml`` file with the following code:
+
+.. code-block:: yaml
+   :caption: config/packages/doctrine_mongodb.yaml
+
+   doctrine_mongodb:
+       auto_generate_proxy_classes: true
+       auto_generate_hydrator_classes: true
+       connections:
+           default:
+               server: '%env(resolve:MONGODB_URL)%&appName=devrel.content.php'
+               options: {}
+       default_database: '%env(resolve:MONGODB_DB)%'
+       document_managers:
+           default:
+               auto_mapping: true
+               mappings:
+                   App:
+                       dir: '%kernel.project_dir%/src/Document'
+                       mapping: true
+                       type: attribute
+                       prefix: 'App\Document'
+                       is_bundle: false
+                       alias: App
+
+Install Frontend Dependency
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This project uses ``twig``, the default template engine for Symfony, to
+generate templates in this application. Run the following command to
+install the ``twig`` bundle:
+
+.. code-block::
+
+   composer require symfony/twig-bundle
+
+.. note::
+
+   This step might result in an error message about unset environment
+   variables, but this is resolved in the following section.
+
+Modify Project Files
+~~~~~~~~~~~~~~~~~~~~
+
+This section demonstrates how to modify the files in your
+``restaurants`` project to create a Symfony web application that views
+restaurants based on filter criteria.
```

Comment:
Is "filter criteria" referring to the query in RestaurantController selecting restaurants in Queens with "Moon" in the name? As a reader, I would have assumed this is actually going to expose some UI to let users dynamically filter the results.

Reply:
I will update the wording. Creating a dynamic filter seemed out of scope for this simple app